src/SFA.DAS.ApprenticeFeedback.Web/Pages/ExitSurvey/Question2.cshtml.cs

```diff
@@ -1,79 +1,173 @@
+            EmployerAttributes = new List<FeedbackAttribute>(await ApprenticeFeedbackService.GetExitSurveyAttributes(ExitSurveyAttributeCategory.Employer));
+            TrainingProviderAttributes = new List<FeedbackAttribute>(await ApprenticeFeedbackService.GetExitSurveyAttributes(ExitSurveyAttributeCategory.TrainingProvider));
+
+            // Load session state if it exists
 
-            IncompletionReason = ExitSurveyContext.IncompletionReason;
+            var selectedPersonalCircumstancesAttributes = ExitSurveyContext.Attributes.Where(a => a.Category == ExitSurveyAttributeCategory.PersonalCircumstances);
+            foreach(var sessionAttribute in selectedPersonalCircumstancesAttributes)
+            {
+                var a = PersonalCircumstancesAttributes.Find(a => a.Id == sessionAttribute.Id);
+                a.Value = true;
+            }
+
+            var selectedEmployerAttributes = ExitSurveyContext.Attributes.Where(a => a.Category == ExitSurveyAttributeCategory.Employer);
+            foreach (var sessionAttribute in selectedEmployerAttributes)
+            {
+                var a = EmployerAttributes.Find(a => a.Id == sessionAttribute.Id);
+                a.Value = true;
+            }
 
+            var selectedTrainingProviderAttributes = ExitSurveyContext.Attributes.Where(a => a.Category == ExitSurveyAttributeCategory.TrainingProvider);
+            foreach (var sessionAttribute in selectedTrainingProviderAttributes)
+            {
+                var a = TrainingProviderAttributes.Find(a => a.Id == sessionAttribute.Id);
+                a.Value = true;
+            }
             return Page();
         }
 
         public IActionResult OnPost()
         {
+            // Make sure something is selected
+
+            var selectedPersonalCircumstancesAttributes = PersonalCircumstancesAttributes.Where(a => a.Value).ToList();
+            var selectedEmployerAttributes = EmployerAttributes.Where(a => a.Value).ToList();
+            var selectedTrainingProviderAttributes = TrainingProviderAttributes.Where(a => a.Value).ToList();
+
+            int selectedCount = selectedPersonalCircumstancesAttributes.Count()
+                + selectedEmployerAttributes.Count()
+                + selectedTrainingProviderAttributes.Count();
+
+            if (0 == selectedCount)
+            {
+                ModelState.AddModelError("MultipleErrorSummary", "Select the factors that contributed to you not finishing your apprenticeship");
+            }
+
             if (!ModelState.IsValid)
             {
                 return Page();
             }
 
-            ExitSurveyContext.IncompletionReason = IncompletionReason;
+            // Decide if something has changed
+            var changedAnswers = false;
+            if (ExitSurveyContext.Attributes.Any(a => a.Category == ExitSurveyAttributeCategory.PersonalCircumstances)
```

Comment:
I know what this is doing but it feels very long winded. 

Can we not simplify it down into something along the lines of

// Get one big list of all the attributes selected from the page as all the Ids will be unique?
selectedPersonalCircumstancesAttributes.Union(selectedEmployerAttributes).Union(selectedProviderAttributes)

Everything in current context that's been removed = Exit SurveyContext.Attributes.Except(Union'dList)

Anything New = Union'dList.Except(ExitSurveyContextAttributes)

If Everything is an empty list, and anything new is an empty list, nothing changed, vs if there is any difference, we force the routing?

Maybe i'm simplifying the logic too much but separating out into the individual categories while works, seems overkill?

Reply:
The attributes are already in separate categories in the session. So this suggestion:
Anything New = Union'dList.Except(ExitSurveyContextAttributes)
won't work because ExitSurveyContextAttributes can potentially contain attributes from other pages.